extension/src/Runner/testRunner.ts

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import { TestSuite } from "../protocols";
+import { RunnerResultStream } from "./runnerResultStream";
+import { ITestRunnerEnvironment } from "./testRunnerEnvironment";
+
+export interface ITestRunner {
+    setup(tests: TestSuite[], isDebugMode: boolean): Promise<ITestRunnerEnvironment>;
+    run(env: ITestRunnerEnvironment): Promise<RunnerResultStream>;
+    updateTestStatus(env: ITestRunnerEnvironment, result: RunnerResultStream): Promise<void>;

```

Comment:
Who will be calling this and what does this function do?

Reply:
I'd like to create a single entry which has a pool of different runners and expose a static method `run` to outside. and this entry would call all the interface methods of `ITestRunner` to run the tests and update the status. It would be like this:
```
const runner: ITestRunner = getMatchedRunner(....);
const env = await runner.setup(tests, isDebug);
await runner.run(env).then((stream) => runner.updateTestStatus(env, stream));
```